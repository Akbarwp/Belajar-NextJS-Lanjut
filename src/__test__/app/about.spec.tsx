import {render, screen} from "@testing-library/react";
import AboutPage from "@/app/about/page";
import AboutLayout from "@/app/about/layout";

describe('About page', () => {
    it('should render', () => {
        const page = render(
            <AboutLayout>
                <AboutPage />
            </AboutLayout>,
        );
        expect(page).toMatchSnapshot();
    });
});