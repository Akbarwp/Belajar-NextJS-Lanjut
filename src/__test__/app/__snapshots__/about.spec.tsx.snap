// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="fixed left-0 top-15 z-10 h-screen w-60 bg-blue"
      >
        <ul
          class="menu menu-sm dropdown-content mt-3 ml-3 z-[1] px-2 py-5 shadow shadow-dark-blue rounded-box w-52"
          tabindex="0"
        >
          <li
            class="text-bone-pink hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
          >
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="text-bone-pink hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
              href="/shop"
            >
              Shop
            </a>
            <ul
              class="p-2"
            >
              <li
                class="text-orange hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
              >
                <a
                  href="/shop/guitar"
                >
                  Guitar
                </a>
              </li>
              <li
                class="text-orange hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
              >
                <a
                  href="/shop/guitar/bass"
                >
                  Bass
                </a>
              </li>
            </ul>
          </li>
          <li
            class="text-bone-pink hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
          >
            <a
              href="/product"
            >
              Product
            </a>
          </li>
        </ul>
      </nav>
      <div>
        <main
          class="flex h-screen items-center justify-center hero"
        >
          <div
            class="hero-content text-center"
          >
            <div
              class="max-w-md"
            >
              <h1
                class="text-5xl font-bold text-bone-pink"
              >
                About Page
              </h1>
              <p
                class="py-6 text-bone-pink"
              >
                Provident cupiditate voluptatem et in. Quaerat fugiat ut assumenda excepturi exercitationem quasi. In deleniti eaque aut repudiandae et a id nisi.
              </p>
              <button
                class="btn btn-primary"
              >
                Get Started
              </button>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="fixed left-0 top-15 z-10 h-screen w-60 bg-blue"
    >
      <ul
        class="menu menu-sm dropdown-content mt-3 ml-3 z-[1] px-2 py-5 shadow shadow-dark-blue rounded-box w-52"
        tabindex="0"
      >
        <li
          class="text-bone-pink hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="text-bone-pink hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
            href="/shop"
          >
            Shop
          </a>
          <ul
            class="p-2"
          >
            <li
              class="text-orange hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
            >
              <a
                href="/shop/guitar"
              >
                Guitar
              </a>
            </li>
            <li
              class="text-orange hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
            >
              <a
                href="/shop/guitar/bass"
              >
                Bass
              </a>
            </li>
          </ul>
        </li>
        <li
          class="text-bone-pink hover:bg-bone-pink/20 hover:rounded-lg hover:transition"
        >
          <a
            href="/product"
          >
            Product
          </a>
        </li>
      </ul>
    </nav>
    <div>
      <main
        class="flex h-screen items-center justify-center hero"
      >
        <div
          class="hero-content text-center"
        >
          <div
            class="max-w-md"
          >
            <h1
              class="text-5xl font-bold text-bone-pink"
            >
              About Page
            </h1>
            <p
              class="py-6 text-bone-pink"
            >
              Provident cupiditate voluptatem et in. Quaerat fugiat ut assumenda excepturi exercitationem quasi. In deleniti eaque aut repudiandae et a id nisi.
            </p>
            <button
              class="btn btn-primary"
            >
              Get Started
            </button>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
